dev/release/verify-release-candidate.sh

```diff
@@ -563,7 +563,10 @@ test_package_java() {
   show_header "Build and test Java libraries"
 
   # Build and test Java (Requires newer Maven -- I used 3.3.9)
-  maybe_setup_conda maven || exit 1
+  # Pin OpenJDK 17 since OpenJDK 20 is incompatible with our versions
+  # of things like Mockito, and we also can't update Mockito due to
+  # not supporting Java 8 anymore
+  maybe_setup_conda maven openjdk=17.0.3 || exit 1
```

Comment:
Won't be also needed to verify also with JDK 8, 11 and 18?

Reply:
The verification script only ever verified with one version of Java before. We can file a separate task if you want to verify for multiple versions within the script.

Reply:
Just filled https://github.com/apache/arrow/issues/35951